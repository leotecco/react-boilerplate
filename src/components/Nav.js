import React from 'react'
import { Link } from 'react-router-dom'
import SignOutButton from './SignOutButton'

const Home = () => (
  <nav>
    <ul>
      <li>
        <Link to="/">Home</Link>
      </li>
      <li>
        <Link to="/count">Count</Link>
      </li>
      <li>
        <Link to="/users">Users</Link>
      </li>
      <SignOutButton />
    </ul>
  </nav>
)

export default Home
