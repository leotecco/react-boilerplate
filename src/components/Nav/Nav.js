import React from 'react'
import { Link } from 'react-router-dom'
import SignOutButton from 'components/SignOutButton'

const Home = () => (
  <nav>
    <ul>
      <li>
        <Link to="/">Home</Link>
      </li>
      <li>
        <Link to="/count">Count</Link>
      </li>
      <li>
        <Link to="/users">Users</Link>
      </li>
      <SignOutButton />
    </ul>
  </nav>
)

export default Home
