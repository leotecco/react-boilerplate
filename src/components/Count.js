import React from 'react'
import { connect } from 'react-redux'
import { increase, decrease } from 'actions'
import Nav from 'components/Nav'

const Count = ({ dispatch, count }) => {
  return (
    <React.Fragment>
      <Nav />
      <h1>Count {count}</h1>
      <button onClick={() => dispatch(increase())}>+</button>
      <button onClick={() => dispatch(decrease())}>-</button>
    </React.Fragment>
  )
}

const mapStateToProps = state => state

export default connect(mapStateToProps)(Count)
