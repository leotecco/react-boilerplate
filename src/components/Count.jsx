import React from 'react'
import { connect } from 'react-redux'
import { INCREASE, DECREASE } from 'constants/count'
import Nav from 'components/Nav'

const Count = ({ dispatch, count }) => {
  return (
    <React.Fragment>
      <Nav />
      <h1>Count {count}</h1>
      <button onClick={() => dispatch(INCREASE)}>+</button>
      <button onClick={() => dispatch(DECREASE)}>-</button>
    </React.Fragment>
  )
}

const mapStateToProps = state => state

export default connect(mapStateToProps)(Count)
