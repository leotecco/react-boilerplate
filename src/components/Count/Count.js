import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import Nav from 'components/Nav'

const Button = styled.button`
  background: transparent;
  border-radius: 3px;
  border: 2px solid #333;
  color: #333;
  font-weight: bold;
  font-size: 1.5rem;
  margin: 0 1em;
  padding: 0.25em 1em;

  &:hover {
    background: #333;
    cursor: pointer;
    color: white;
  }
`

const Title = styled.h1`
  font-family: sans-serif;
  color: #333;
`

export const Count = ({ total, increase, decrease }) => {
  return (
    <React.Fragment>
      <Nav />
      <Title>Total: {total}</Title>
      <Button onClick={increase} id="increase">
        +
      </Button>
      <Button onClick={decrease} id="decrease">
        -
      </Button>
    </React.Fragment>
  )
}

Count.propTypes = {
  total: PropTypes.number.isRequired,
  increase: PropTypes.func.isRequired,
  decrease: PropTypes.func.isRequired,
}

export default Count
