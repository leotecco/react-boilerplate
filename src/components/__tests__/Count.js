import { Count } from '../Count'
import Nav from 'components/Nav'

describe('Count Component', () => {
  let wrapper, total, increase, decrease
  beforeEach(() => {
    total = 1
    increase = jest.fn()
    decrease = jest.fn()
    const props = { total, increase, decrease }
    wrapper = shallow(<Count {...props} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the Nav Component', () => {
    expect(wrapper.find(Nav)).toBeTruthy()
  })

  it('should invoke increase when click in the button', () => {
    const increaseButton = wrapper.find('#increase')
    increaseButton.simulate('click')
    expect(increase).toHaveBeenCalled()
  })

  it('should invoke decrease when click in the button', () => {
    const decreaseButton = wrapper.find('#decrease')
    decreaseButton.simulate('click')
    expect(decrease).toHaveBeenCalled()
  })
})
