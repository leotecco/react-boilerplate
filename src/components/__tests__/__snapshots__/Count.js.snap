// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Count Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Count
    decrease={[MockFunction]}
    increase={[MockFunction]}
    total={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Home />,
        <h1>
          Total: 
          1
        </h1>,
        <button
          id="increase"
          onClick={[MockFunction]}
        >
          +
        </button>,
        <button
          id="decrease"
          onClick={[MockFunction]}
        >
          -
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Total: ",
            1,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Total: ",
          1,
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+",
          "id": "increase",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "+",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "-",
          "id": "decrease",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "-",
        "type": "button",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Home />,
          <h1>
            Total: 
            1
          </h1>,
          <button
            id="increase"
            onClick={[MockFunction]}
          >
            +
          </button>,
          <button
            id="decrease"
            onClick={[MockFunction]}
          >
            -
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Total: ",
              1,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Total: ",
            1,
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+",
            "id": "increase",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "+",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "-",
            "id": "decrease",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "-",
          "type": "button",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
